src/main/java/org/jabref/gui/theme/SystemThemeDetector.java

```diff
@@ -0,0 +1,121 @@
+package org.jabref.gui.theme;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+
+import org.jabref.architecture.AllowedToUseStandardStreams;
+
+// This class is extracted from https://gist.github.com/HanSolo/7cf10b86efff8ca2845bf5ec2dd0fe1d.
+// Permission has been explicitly granted by the author.
+
+@AllowedToUseStandardStreams("Required to detect dark theme in Windows OS")
+public class SystemThemeDetector {
+    public enum OperatingSystem {
+        WINDOWS, MACOS, LINUX, SOLARIS, NONE
+    }
+
+    private static final String REGQUERY_UTIL = "reg query ";
+    private static final String REGDWORD_TOKEN = "REG_DWORD";
+    private static final String DARK_THEME_CMD = REGQUERY_UTIL + "\"HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize\"" + " /v AppsUseLightTheme";
+
+    public final boolean isDarkMode() {
+        switch (getOperatingSystem()) {
+            case WINDOWS: return isWindowsDarkMode();
+            case MACOS: return isMacOsDarkMode();
+            // currently theme detection not implemented for Linux and Solaris
+            case LINUX: return false;
+            case SOLARIS: return false;
+            default: return false;
+        }
+    }
```

Comment:
Can you please split up this class and use the existing OS package and classes to provide a method isDarkMode instead of introducing some code duplication to test for which os is runnning here?